README.md

```diff
@@ -681,37 +681,57 @@ non-IO.
 
 # Developing Semian
 
-Semian requires a Linux environment.  We provide a [Vagrantfile](Vagrantfile)
-that installs MySQL, Redis and Ruby in an Ubuntu 16.04 virtual machine.  Use
+Semian requires a Linux environment.  We provide a [docker-compose](https://docs.docker.com/compose/) file
+that runs MySQL, Redis, Toxiproxy and Ruby in containers.  Use
 the steps below to work on Semian from a Mac OS environment.
 
+Prerequisites :
 ```bash
-# install Vagrant and VirtualBox
-$ brew cask install vagrant virtualbox
+# install docker-for-desktop
+$ brew cask install docker
+
+# install visual-studio-code (optional)
+$ brew cask install visual-studio-code
 
 # clone Semian
 $ git clone https://github.com/Shopify/semian.git
 $ cd semian
+```
 
-# build the virtual machine (can take a while for ruby to compile)
-$ vagrant up
+Developing :
 
-# shell in to the virtual machine
-$ vagrant ssh
+- Visual Studio Code
+  - Open semian in vscode
+  - Install recommended extensions (one off requirement)
+  - Click `reopen in container` (first boot might take about a minute)
+  
+  See https://code.visualstudio.com/docs/remote/containers for more details
 
-# start the toxiproxy-server inside the virtual machine
-$ toxiproxy-server > /dev/null &
-# TODO: provide a systemd unit for toxiproxy
+- Everything else
+  - `$ docker-compose -f .devcontainer/docker-compose.yml up -d`
+  - `$ docker exec -it semian-dev bash`
+  - `$ cd /workspace && bundle install`
+  -  Use vim or your TUI editor of choice inside the container 
+  
+  OR
+  
+  - Make changes outside the container and run tests inside the container
+
+Running Tests:
+  - `$ docker-compose -f .devcontainer/docker-compose.yml up -d`
+  - `$ docker exec -it semian-dev bash`
+  - `$ cd /workspace && bundle install`
+  - `$ bundle exec rake`
+
+
+If you make any changes to `.devcontainer/..`, you'd need to recreate the containers:
+  - `$ docker-compose -f .devcontainer/docker-compose.yml up -d --force-recreate` 
+
+
+  OR
```

Comment:
nit: All of these `OR` instructions complicates the flow. It may be better organized into two sections: one dedicated to VSCode and another for everyone else.